### PR TITLE
refactor(redis-clients): create an abstraction for all clients

### PR DESCRIPTION
**Pull request by sileht**
This introduces a new object RedisLinks that hold and lazy load
reference to all Redis clients.

Fixes MRGFY-1113

### CODE REVIEW

File: mergify_engine/config.py [hunk 1]
```diff
@@ -218,9 +218,9 @@ def ApplicationAPIKeys(v: str) -> typing.Dict[str, ApplicationAPIKey]:
             "TESTING_FORWARDER_ENDPOINT",
             default="https://test-forwarder.mergify.io/events",
         ): str,
-        voluptuous.Required("INSTALLATION_ID", default=15398551): voluptuous.Coerce(
-            int
-        ),
+        voluptuous.Required(
+            "TESTING_INSTALLATION_ID", default=15398551
+        ): voluptuous.Coerce(int),
```
Comment (jd) [inline mergify_engine/config.py:223]: unrelated?

File: mergify_engine/redis_utils.py [hunk 2]
```diff
@@ -13,17 +13,28 @@
...
+import yaaredis
 
-from mergify_engine import utils
+from mergify_engine import config
 
 
 LOG = daiquiri.getLogger(__name__)
 
+
+# NOTE(sileht): I wonder with mypy thing yaaredis.StrictRedis is Any...
```
Comment (jd) [inline mergify_engine/redis_utils.py:33]:
```suggestion
# NOTE(sileht): I wonder why mypy thinks `yaaredis.StrictRedis` is `Any`...
```

File: mergify_engine/web/api/security.py [hunk 3]
```diff
@@ -100,11 +100,8 @@ async def get_repository_context(
     repository: github_types.GitHubRepositoryName = fastapi.Path(  # noqa: B008
         ..., description="The name of the repository"
     ),
-    redis_cache: utils.RedisCache = fastapi.Depends(  # noqa: B008
-        redis.get_redis_cache
-    ),
-    redis_queue: utils.RedisQueue = fastapi.Depends(  # noqa: B008
-        redis.get_redis_queue
+    redis_links: redis_utils.RedisQueue = fastapi.Depends(  # noqa: B008
```
Comment (Syffe) [inline mergify_engine/web/api/security.py:103]:
```suggestion
    redis_links: redis_utils.RedisLinks = fastapi.Depends(  # noqa: B008
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/config.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/config.py:223]: I typed all tests using redis_XXX and some was complaining about missing types, when I added the type for INSTALLATION_ID I prefixed it with TESTING_ as it's testing only thing.

File: mergify_engine/web/api/security.py (hunk 3 above)
Comment (sileht) [reply mergify_engine/web/api/security.py:103]: Hum, wierd that mypy didn't cached that one...

**Review by jd — APPROVED** [review]
